### PR TITLE
Remove chiquito2halo2 function in halo2 backend

### PR DESCRIPTION
**Pull request by alxkzmn**

### CODE REVIEW

File: src/plonkish/compiler/mod.rs [hunk 1]
```diff
@@ -54,11 +54,20 @@ pub fn compile<
     }
 }
 
-pub struct PlonkishCompilationResult<F, TG> {
+pub struct PlonkishCompilationResult<F: Clone, TG: TraceGenerator<F> + Clone> {
     pub circuit: Circuit<F>,
     pub assignment_generator: Option<AssignmentGenerator<F, TG>>,
 }
 
+impl<F: Clone, TG: TraceGenerator<F> + Clone> Clone for PlonkishCompilationResult<F, TG> {
```
Comment (leolara) [inline src/plonkish/compiler/mod.rs:62]: @alxkzmn I think you can do an automatic derivation

File: src/plonkish/compiler/mod.rs [hunk 2]
```diff
@@ -54,11 +54,20 @@ pub fn compile<
     }
 }
 
-pub struct PlonkishCompilationResult<F, TG> {
+pub struct PlonkishCompilationResult<F: Clone, TG: TraceGenerator<F> + Clone> {
```
Comment (leolara) [inline src/plonkish/compiler/mod.rs:57]: @alxkzmn I don't think you need to make the generic types Clone here, just in the impl clause that needs that

**Comment by leolara** [comment]
@alxkzmn

now the halo2 mock prover is not a mock prover, anymore?

The changes to python, apart random setup, are just linting?

**Comment by alxkzmn** [comment]
Yep, it's not a mock prover anymore but I didn't want to rename it everywhere because we're still waiting for https://github.com/privacy-scaling-explorations/halo2/pull/356 to be merged so we can have the "mock prover" again. Those big changes to Poseidon example is because of formatting, the only real change there is passing the `params_path` value.

**Review by leolara — APPROVED** [review]

**Pull request merged**